modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPInstanceMVCActionCommand.java

```diff
@@ -219,7 +220,25 @@ private void _buildCPInstances(ActionRequest actionRequest)
 		ServiceContext serviceContext = ServiceContextFactory.getInstance(
 			CPInstance.class.getName(), actionRequest);
 
-		_cpInstanceService.buildCPInstances(cpDefinitionId, serviceContext);
+		List<CPInstance> cpInstances = _cpInstanceService.buildCPInstances(
```

Comment:
I think this should be put in a callable as well. Can you please review this and advise @alessiorendina @luca-pellizzon , thanks!

Reply:
Aren't services already transactional? If so the database should be untouched when any exception occurs.